libs/usb-drive/src/usb_drive.ts

```diff
@@ -0,0 +1,143 @@
+    return { ...assertDefined(rawData.blockdevices[0]), path: devicePath };
+  } catch (error) {
+    debug(`Error getting block device info for ${devicePath}: ${error}`);
+    return undefined;
+  }
+}
+
+async function findUsbDriveDevice(): Promise<string | undefined> {
+  const DEVICE_PATH_PREFIX = '/dev/disk/by-id/';
+  const USB_REGEXP = /^usb(.+)part(.*)$/;
+
+  // List devices, filtering to only the USB partitions
+  const devicesById = (await fs.readdir(DEVICE_PATH_PREFIX)).filter((name) =>
+    USB_REGEXP.test(name)
+  );
+
+  // We only support one USB drive at a time
+  const [deviceId] = devicesById;
+  if (!deviceId) {
+    return undefined;
+  }
+
+  // Follow the symlink
+  const devicePath = join(
+    DEVICE_PATH_PREFIX,
+    await fs.readlink(join(DEVICE_PATH_PREFIX, deviceId))
+  );
+
+  return devicePath;
+}
+
+async function getUsbDriveStatus(): Promise<BlockDeviceInfo | undefined> {
+  const devicePath = await findUsbDriveDevice();
+  if (!devicePath) return undefined;
+  debug(`Found USB drive at ${devicePath}`);
+  return await getBlockDeviceInfo(devicePath);
+}
+
+const VX_MOUNT_POINT = '/media/vx/usb-drive';
+
+async function mountUsbDrive(devicePath: string): Promise<void> {
+  debug(`Mounting USB drive ${devicePath} at ${VX_MOUNT_POINT}`);
+  await fs.mkdir(VX_MOUNT_POINT, { recursive: true });
+  await exec('mount', [
+    '-w',
+    '-o',
+    'umask=000,nosuid,nodev,noexec',
+    devicePath,
+    VX_MOUNT_POINT,
+  ]);
+}
+
+async function unmountUsbDrive(mountPoint: string): Promise<void> {
+  debug(`Unmounting USB drive at ${mountPoint}`);
+  await exec('umount', [mountPoint]);
+}
+
+export function detectUsbDrive(): UsbDrive {
+  let didEject = false;
+
+  return {
+    async status(): Promise<UsbDriveStatus> {
+      let deviceInfo = await getUsbDriveStatus();
+      if (!deviceInfo) return { status: 'no_drive' };
+      // Automatically mount the drive if it's not already mounted
+      // TODO if the drive is mounted at a different mount point than our
+      // default VX mount point, do we want to unmount and remount it?
+      if (!deviceInfo.mountpoint && !didEject) {
+        await mountUsbDrive(deviceInfo.path);
+        deviceInfo = assertDefined(await getUsbDriveStatus());
+      }
+      if (deviceInfo.mountpoint) {
+        return { status: 'mounted', mountPoint: deviceInfo.mountpoint };
+      }
+      return { status: 'ejected' };
+    },
+
+    async eject(): Promise<void> {
+      const deviceInfo = await getUsbDriveStatus();
+      if (!deviceInfo) {
+        throw new Error('No USB drive detected');
+      }
+      if (!deviceInfo.mountpoint) {
+        throw new Error('USB drive is not mounted');
+      }
+      await unmountUsbDrive(deviceInfo.mountpoint);
+      didEject = true;
+    },
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    async format(): Promise<void> {
+      throw new Error('Not implemented');
+    },
+  };
+}
```

Comment:
I omitted the logic about unmounting phantom USB drives: https://github.com/votingworks/kiosk-browser/blob/main/src/ipc/get-usb-drive-info.ts#L83. Is that ok?

Reply:
When I first moved the mounting logic from `kiosk-browser` to a script called by `kiosk-browser`, I omitted this step and then found it to be problematic. The case comes up quite a lot and, as an OS problem, I think the way we were handling is the best we can do.

Just diving into a review now.

Reply:
Discussed and decided to leave this bit out for now until we can repro the problem and understand better the underlying issue.